python/lib/communication/dmod/communication/maas_request/ngen/partial_realization_config.py

```diff
@@ -0,0 +1,68 @@
+from ngen.config.configurations import Routing
+from ngen.config.formulation import Formulation
+from ngen.config.realization import CatchmentRealization
+from pydantic import BaseModel, validator
+from typing import ClassVar, Dict, List, Optional
+
+
+class PartialRealizationConfig(BaseModel):
+    """
+    Helper class for working with the serialized formulation configurations ::class:`AbstractNgenRequest` messages.
+
+    The type relies on classes from the ``ngen-config`` external package and the transitive ``pydantic`` dependency.
+
+    Note that this type has two optional forcing-config-related fields. These are expected to be used infrequently, as
+    generally DMOD will automatically determine the correct forcing dataset. They do allow for optional user control,
+    but the more specific and essential need (which does rely on such control being available) is to support a
+    workaround for reading forcing data directly from the Docker host via a bind mount. This functionality is handled by
+    the scheduling logic, but the generated realization config must properly reflect it.
+
+    The workaround is triggered by adding specific prefix (::attribute:`_FROM_ENV_PREFIX`) and delimiter
+    (::attribute:`_FROM_ENV_DELIMIT`) substrings to either of the forcing config fields.
+    """
+
+    _FROM_ENV_DELIMIT: ClassVar[str] = ':::'
+    _FROM_ENV_PREFIX: ClassVar[str] = 'from_env'
+
+    hydrofabric_uid: str
```

Comment:
Should we consider enforcing this as a `UUID` type?

Reply:
The `hydrofabric_uid` is the id value generated by the hydrofabric itself.  It needs to be unique and _deterministic_:  essentially a hash value for the hydrofabric's data and metadata.  

I suppose we still _could_ use _UUID_ for it.  It just seems a little more tricky and a little less beneficial, given we aren't generating a random unique value.